src/components/ha-icon-picker.ts

```diff
@@ -167,7 +167,10 @@ export class HaIconPicker extends LitElement {
       filteredItems.push(...filteredItemsByKeywords);
 
       if (filteredItems.length > 0) {
-        this.comboBox.filteredItems = filteredItems;
+        // Setting it sync cause browser freeze when user paste an icon text
+        setTimeout(() => {
+          this.comboBox.filteredItems = filteredItems;
```

Comment:
I think this is a dirty fix to a bigger problem.  The problem I think stems from going behind lit's back to update the items instead of tying them to a reactive property.  So whenever a litElement update is triggered, you are starting over with `iconItems` because that's in `render`.

Instead, `@property` the filtered items, update them on input, and let lit handle the update.

Reply:
Yes you right. I think the filtered items assignment is old code (created before ha-combobox) and doesn't make sense anymore. I will look to refactor the component 🙂

Reply:
👍 I think you could probably keep most as is and just replace `this.combobox.filteredItems` everywhere with a reactive property update.  Might also need to call `this.requestUpdate()` if the property is only mutated.

Reply:
I changed it to a reactive state but the setTimeout is always needed... 😕 I have the same behavior

Reply:
Well, at least it's cleaner now 😜 

I was so in disbelief that this actually fixes the problem that I pulled the branch down to try it out.  When I test, the initial paste into an empty field goes smoothly, but then I can repeatedly crash the browser by:

1. Tab to the next field
2. Shift+tab back to the icon picker
3. Paste over the 1st paste or just delete it and paste again

I'll try to spend a few more minutes on it to see if I can narrow it down.

Reply:
It also depend of the content paste. If you paste `mdi:home`, it crash. If you paste `mdi:account` it doesn't crash. I think the proper solution is to use web worker for the filter to do the search in a background thread.